### PR TITLE
New way to check classic vms

### PR DESCRIPTION
**Pull request by jesusbv**
- Add new way to check instances running in classic
  because previous query doesn't produce HTTP error
  any more. Classic vms do not have instance data.

### CODE REVIEW

File: azuremetadata [hunk 1]
```diff
@@ -52,15 +52,15 @@ try:
     # internal server error, that's the best thing we have to go on to
     # figure out whether or not we are in ASM.
     # ASM gets retired in 2023, rip this code out, it's ugly!
-    attested_data_url = 'http://169.254.169.254/metadata/attested/document'
+    attested_data_url = 'http://169.254.169.254/metadata/instance/compute'
```
Comment (ikapelyukhin) [inline azuremetadata:55]: Does `/metadata/instance/compute` also return the attested data?

**Review by ikapelyukhin — APPROVED** [review]
Code LGTM, but I haven't verified whether `/metadata/instance/compute` returns attested data or not (in the past I think it didn't).

File: azuremetadata (hunk 1 above)
Comment (jesusbv) [reply azuremetadata:55]: No, `/metadata/instance/compute` does not return attested data

It returns the info in [sample 4 response](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=linux#sample-4-get-more-information-about-the-vm-during-support-case) or raise a HTTP error if vm is `classic`

The reason for the change is that, attested data endpoint does not produce an error in `classic` anymore
but `metadata/instance/compute` does

**Comment by jesusbv** [comment]
> That's what I have locally, but no answer from Msft whether this will continue to be true....

If it will not, we need to find a new way to check the instance is `classic`

**Review by rjschwei — APPROVED** [review]

File: azuremetadata (hunk 1 above)
Comment (ikapelyukhin) [reply azuremetadata:55]: OK, so there's a variable called `attested_data_url` which never returns any attested data (which we expect to have on RMT side for verification of non-classic VMs). Confused I am.

**Pull request merged**